infra/service-bus/namespace.bicep

```diff
@@ -0,0 +1,67 @@
+@description('Required. The parameter object for the virtual network. The object must contain the name,resourceGroup and subnetPrivateEndpoints values.')
+param vnet object
+
+@description('Required. The parameter object for servicebus. The object must contain the namespaceName,namespacePrivateEndpointName and skuName values.')
+param serviceBus object
+
+@description('Required. The Azure region where the resources will be deployed.')
+param location string = resourceGroup().location
+
+@description('Required. Environment name.')
+param environment string
+
+@description('Required. Date in the format yyyyMMdd-HHmmss.')
+param deploymentDate string = utcNow('yyyyMMdd-HHmmss')
+
+@description('Required. Date in the format yyyy-MM-dd.')
+param createdDate string = utcNow('yyyy-MM-dd')
+
+var customTags = {
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+}
+var tags = union(loadJsonContent('../default-tags.json'), customTags)
+
+var serviceBusTags = {
+  Name: serviceBus.namespaceName
+  Purpose: 'Service Bus Namespace'
+  Tier: 'Shared'
+}
+
+var serviceBusPrivateEndpointTags = {
+  Name: serviceBus.namespacePrivateEndpointName
+  Purpose: 'Service Bus Namespace private endpoint'
+  Tier: 'Shared'
+}
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2023-02-01' existing = {
+  scope: resourceGroup(vnet.resourceGroup)
+  name: vnet.name
+  resource privateEndpointSubnet 'subnets@2023-02-01' existing = {
+    name: vnet.subnetPrivateEndpoints
+  }
+}
+
+module serviceBusResource 'br/SharedDefraRegistry:service-bus.namespaces:0.5.7' = {
+  name: 'service-bus-${deploymentDate}'
+  params: {
+    name: serviceBus.namespaceName
+    skuName: serviceBus.skuName
+    location: location
+    diagnosticWorkspaceId: ''
+    lock: 'CanNotDelete'
+    networkRuleSets: {
+      publicNetworkAccess: 'Disabled'
+    }
+    privateEndpoints: [
+      {
+        name: serviceBus.namespacePrivateEndpointName
+        service: 'namespace'
+        subnetResourceId: virtualNetwork::privateEndpointSubnet.id
```

Comment:
Can we use the resourceId function here?

Reply:
@ajaysoni-cap I went with this approach based on a few articles i'd read and we're also using the same approach for AKS.

This taken from https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/best-practices .  This is referring more to using 'existing' over 'outputs'

Instead of passing property values around through outputs, use the [existing keyword](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/existing-resource) to look up properties of resources that already exist. It's a best practice to look up keys from other resources in this way instead of passing them around through outputs. You'll always get the most up-to-date data.

This taken from https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-resource.  This I think is referring to another resource already deployed in the same bicep file, but again we could apply the same concept using the 'existing' function.

The resourceId function is available in Bicep files, but typically you don't need it. Instead, use the symbolic name for the resource and access the id property.

Happy to change it to use 'resourceId' if that's the way we want to go.  Better to make that decision now before we start adding more resources because we only have a couple of resources deployed at the moment, it should be easy to swap out for resourceId and consistently use this approach going forward or we stick with 'existing' and go with that approach.

@ajaysoni-cap @DaRoszko @ppotturi @ganeshg2491  What do you guys think?

Reply:
Prefer the resourceId, has fewer lines of code

subnetResourceId: resourceId(vnet.resourceGroup, 'Microsoft.Network/virtualNetworks/subnets', vnet.name, vnet.subnetPrivateEndpoints)

Reply:
Thanks @kbabigumira . I'll change to resourceId, unless someone has any objections and sorry I thought I added you in the @ list when I asked the question.

Reply:
existing is definitely preferred over the outputs and we must avoid using functions where resources are defined within the same template but in this scenario, we don't have either of these.